torch/csrc/jit/passes/shape_analysis.cpp

```diff
@@ -150,6 +150,7 @@ void PropagateShapeOnNode(Node * node, bool insert_expands) {
   }
   std::vector<TensorType*> types;
   bool present;
+  // TODO (apaszke): This currently assumes that inputs are pretty much tensor-only. Need to fix that.
```

Comment:
I have partial fixes for this, but I only workaround it by not propagating when an input is not a tensor. We will need to refactor a bit so that shape prop works with int/float.